### DISCUSSION TITLE
Code doesn't produce diagnostic plot

### DISCUSSION THREAD

**Discussion by LinRoman** [georgebv/pyextremes#22] [Q&A] [answered]
Hi George

I'm testing the code with a much shorter time series than the example input data. I sampled the extreme values with a block_size="50D" and it seems to work.

[F6_Lucens.csv](https://github.com/georgebv/pyextremes/files/8232468/F6_Lucens.csv)

![image](https://user-images.githubusercontent.com/101408776/157881892-f70aab81-e19b-41ac-a3d0-e0c59e8d4914.png)

I can get a model summary with model.get_summary(), but the code fails to produce the graphics with model.plot_diagnostic(alpha=0.95).

May I ask you if you've got a hint why it doesn't work?

Thanks very much for your help!

Below I paste the code I used:

import pandas as pd
import numpy as np
import pyextremes
from pyextremes import EVA
from pyextremes import get_extremes, get_return_periods
import matplotlib.pyplot as plt

data = pd.read_csv(
    "F6_Lucens.csv",
    index_col=0,
    parse_dates=True,
    squeeze=True,
)

data = (
    data
    .sort_index(ascending=True)
    .astype(float)
    .dropna()
)
#data = data.loc[pd.to_datetime("1925"):]
#data = data - (data.index.array - pd.to_datetime("1992")) / pd.to_timedelta("365.2425D") * 2.87e-3

data.plot()

model = EVA(data=data)
help(EVA.get_extremes)

model.get_extremes(
    method="BM",
    extremes_type="high",
    block_size="50D",
    errors="raise",
)
model
print(model)

model.plot_extremes()

help(EVA.get_extremes)

model.fit_model()
model
print(model)

extremes = get_extremes(
    ts=data,
    method="BM",
    block_size="200D",
)

print(extremes)

return_periods = get_return_periods(
    ts=data,
    extremes=extremes,
    extremes_method="BM",
    extremes_type="high",
    block_size="200D",
    return_period_size="200D",
    plotting_position="weibull",
)
return_periods.sort_values("return period", ascending=False).head()
print(return_periods)

summary = model.get_summary(
     return_period=[0.3,0.5,1, 2, 5, 10,100],
     alpha=0.95,
     n_samples=21,
)
print(summary)

model.plot_return_values(
    return_period=np.logspace(1, 2, 10),
    return_period_size="200D",
    alpha=0.95,
)

model.plot_diagnostic(alpha=0.95)

summary = model.get_summary(
     return_period=[1, 2, 5, 10, 25, 50, 100, 250, 500, 1000],
     alpha=0.95,
     n_samples=1000,
 )
print(summary)

**Comment by georgebv** [comment, marked as answer]
@LinRoman I have created a gist here: https://gist.github.com/georgebv/9db126f6200fd3f577b2d99b7a4a6a60. As you can see, everything is working on my end (Windows 10)

I have noticed that you set `return_period_size` to 200 days and I wanted to make sure you understand what this means - a return period 100 now means 100 x 200 days ~ 55 years, not 100 years.

**Reply by LinRoman** [reply]
Thanks for your answer. Yes, I intentionally changed the size of the return size period :)

Somehow I can't get the code working neither in Jupyter nor Spyder. It keeps stopping as soon as I want to get the results. 

In your e-mail you referred to an option to run the code under a if __name__ == '__main__' block for multiprocessing. To be honest I don't really understand where to insert this or modify the code. Can you give a hint how to do this?

Thanks in advance!

Cheers,
Roman

**Reply by georgebv** [reply]
@LinRoman I suggest you read about this here https://www.freecodecamp.org/news/if-name-main-python-example/.

If you run this in a python file (not jupyter) you need to do the following:

```python
def main():
    # write your code here

if __name__ == "__main__":
    main()
```

```shell
python -m your-script-file.py
```

**Reply by LinRoman** [reply]
Wooah! It's working! :) Thanks very much!
